### PR TITLE
[SPARK-23044] Error handling for jira assignment

### PR DESCRIPTION
**Pull request by squito**
## What changes were proposed in this pull request?

* If there is any error while trying to assign the jira, prompt again
* Filter out the "Apache Spark" choice
* allow arbitrary user ids to be entered

## How was this patch tested?

Couldn't really test the error case, just some testing of similar-ish code in python shell.  Haven't run a merge yet.

### CODE REVIEW

**Comment by squito** [comment]
@jerryshao this should fix it, but I don't have anything to merge to test this out -- would appreciate if someone could try it before we merge.

**Review by srowen — APPROVED** [review]
Yeah, that's about all we can do here. I'm not sure if there's an API for the /roles screen, and adding the user as a contributor, but it's rare enough that it's OK to handle manually.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by vanzin** [comment]
Since you're here, would you like to try at add SPARK-23031 to this change?

**Comment by vanzin** [comment]
Yeah, arbitrary name is fine. I tried it yesterday and the script just yelled at me because it was not an integer. (Filtering out "Apache Spark" is bonus.)

**Comment by jerryshao** [comment]
@squito thanks for the fix. I also don't have PRs to verify the changes, but I think catching exception should be enough.

**Review by felixcheung — APPROVED** [review]

**Comment by squito** [comment]
@vanzin @jerryshao want to take another look?

Now it
* filters out "apache spark"
* lets you enter an arbitrary id
* if there's an error, just prompts again

sample session: https://gist.github.com/squito/de73fbd0b9c00961377068b91283e04c

File: dev/merge_spark_pr.py [hunk 1]
```diff
@@ -298,24 +299,36 @@ def choose_jira_assignee(issue, asf_jira):
...
+                print("[%d] %s (%s)" % (idx, author.displayName, ",".join(annotations)))
+            raw_assignee = raw_input("Enter number of user, or userid,  to assign to (blank to leave unassigned):")
+            if raw_assignee == "":
+                return None
+            else:
+                try:
+                  id = int(raw_assignee)
+                  assignee = candidates[id]
+                except:
+                  # assume its a user id, and try to assign (might fail, then we just prompt again)
```
Comment (vanzin) [inline dev/merge_spark_pr.py]: it's

**Review by vanzin — COMMENTED** [review]
Looks good but you have some style check failures.

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**